### PR TITLE
Details page

### LINKED ISSUE #36: College details page
### Description
Clicking on a college from anywhere within the website would bring up this page that shows details about a given college

### User Story
As prospective student, I want to view details for a selected college so I can figure out if it fits me or not

### Acceptance Criteria

- [ ] Page displays all the same info from card view
- [ ] User can apply from this page
- [ ] Any missing data is indicated by “Not available” 
- [ ] Page has 5 additional sections: Outcomes, Admissions, Student Body, Academics, Costs
- [ ] There are some conditional labels:
      - [ ] **Graduation rate**: If highest degree offered is Associates or below, show <4 year rate. If bach or above, show 4 year rate.
            - [ ] Note that the descriptive text also has a conditional change, noted in the mocks
      - [ ] **Retention**: If highest degree offered is Associates or below, show <4 year rate. If bach or above, show 4 year rate.
      - [ ] **SAT Scores**: The descriptive text indicates whether the SAT is required or not, noted in the mocks
      - [ ] **Average net price by HH income**: This will be a 2-column table like this one (with the same bands on the left)
Image

### Resources
We're finalizing these but should be able to pull into Ready soon. See comments below + Figma.
 
### Definition of Done 
[Definition of Done](https://github.com/coforma/swift-tech-challenge/blob/main/documentation/product/definition_of_done.md)

### PR DESCRIPTION
**Pull request by gmrabian**
# Summary

Still need help with the styling and some formatting

- Add router for clicking on college cards to get to details page

Fixes #36 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How To Test
- Export AWS credentials in your terminal
- Run `yarn dev`
- Open localhost:3000 and click on any college card
- Ensure it navigates to a new page with the college id as the path
- Ensure it shows information relevant to that college's data in the display

# Checklist:

- [x] My code follows the style guidelines of this project and has no linting errors
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation, including any applicable ADRs
- [ ] My changes generate no new warnings
- [ ] I have added tests that fail without these changes
- [ ] New and existing tests (unit, integration, accessibility) pass locally
- [ ] Documentation updated
- [ ] If there are security concerns they are addressed or ticketed after being discussed

### CODE REVIEW

File: src/app/[id]/page.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  CollegeCard,
+  DetailsCards,
+  InstitutionContext,
+} from "@/src/app/components";
+// types
+import { College } from "../types";
+
+const filterCollege = (institutionData: College[], id: number) => {
+  return institutionData.filter((college) => college.id == id)[0];
+};
```
Comment (ntsummers1) [inline src/app/[id]/page.tsx:14]: Is this used anywhere?

**Review by ntsummers1 — APPROVED** [review]
One question I had that I might be mistaken on. Otherwise this looks really good! Nice work

File: src/app/[id]/page.tsx (hunk 1 above)
Comment (gmrabian) [reply src/app/[id]/page.tsx:14]: yes this gets used on line 25! it filters from the total dataset 
```js
useEffect(() => {
    if (institutionData) {
      setSelectedCollege(filterCollege(institutionData!, params.id));
    }
  }, [institutionData, params.id]);
```

Comment (ntsummers1) [reply src/app/[id]/page.tsx:14]: Ah, I'm blind. Nice!

**Review by ntsummers1 — APPROVED** [review]
LGTM! Styles incoming :D

**Pull request merged**